### PR TITLE
PCX-3740 Added jacoco.gradle script for code coverage reporting.

### PR DESCRIPTION
**Pull request by izpakla**
## Jira ticket
[PCX-3740](https://optile.atlassian.net/browse/PCX-3740)

## Overview/What
Add jacoco gradle script to Github repository for generating code-coverage statistics

## Cause/Why
Increase code coverage.

## Solution
Create a quick summary for current code coverage using Jacoco and Github actions.

## Type of change
- New feature (non-breaking change which adds functionality)

### CODE REVIEW

File: .github/workflows/coverage.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Generate coverage report
+        run: ./gradlew generateCoverageReportForCheckout
+      - name: Load code coverage report
+        id: jacoco_reporter
+        uses: PavanMudigonda/jacoco-reporter@v4.6
+        with:
+          coverage_results_path: checkout/build/reports/jacoco/jacocoTestReport/jacocoTestReport.xml
+          coverage_report_name: Coverage
+          coverage_report_title: JaCoCo
+          github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (siebeprojects) [inline .github/workflows/coverage.yml:29]: Does this give both read and write permissions?

Comment (Sorix) [reply .github/workflows/coverage.yml:29]: @siebeprojects could you check that both repositories (Android and iOS) are configured to `Read` default permission: https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#configuring-the-default-github_token-permissions

File: .github/workflows/coverage.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+name: Coverage report
+on:
```
Comment (Sorix) [inline .github/workflows/coverage.yml:2]:
```suggestion
name: Coverage report

on:
```

File: .github/workflows/coverage.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+  workflow_dispatch:
+permissions:
```
Comment (Sorix) [inline .github/workflows/coverage.yml:4]:
```suggestion
  workflow_dispatch:

permissions:
```

File: .github/workflows/coverage.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+  contents: read
+jobs:
```
Comment (Sorix) [inline .github/workflows/coverage.yml:6]:
```suggestion
  contents: read

jobs:
```

File: .github/workflows/coverage.yml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+  create-coverage:
+    runs-on: ubuntu-latest
+    steps:
```
Comment (Sorix) [inline .github/workflows/coverage.yml:9]:
```suggestion
    runs-on: ubuntu-latest

    steps:
```

**Review by Sorix — COMMENTED** [review]
I think it's better to use one styling for GitHub actions. I looked at https://github.com/optile/checkout-android/blob/develop/.github/workflows/app_live_upload_examplecheckout.yml for example and it has different styling.

File: .github/workflows/coverage.yml (hunk 1 above)
Comment (izpakla) [reply .github/workflows/coverage.yml:29]: @siebeprojects In permissions section I only allowed read.
`permissions:
  contents: read`

File: build.gradle [hunk 6]
```diff
@@ -3,6 +3,7 @@ apply from: 'version.gradle'
 apply from: 'browserstack.gradle'
 
 buildscript {
+    ext.jacocoVersion = '0.8.7'
```
[the remark this answers is not in the document]
Comment (izpakla) [reply build.gradle]: @siebeprojects Unified naming.

File: .github/workflows/coverage.yml (hunk 2 above)
Comment (izpakla) [reply .github/workflows/coverage.yml]: Fixed.

File: .github/workflows/coverage.yml (hunk 3 above)
Comment (izpakla) [reply .github/workflows/coverage.yml]: Fixed.

File: .github/workflows/coverage.yml (hunk 4 above)
Comment (izpakla) [reply .github/workflows/coverage.yml]: Fixed.

File: .github/workflows/coverage.yml (hunk 5 above)
Comment (izpakla) [reply .github/workflows/coverage.yml]: Fixed.

**Review by Sorix — APPROVED** [review]

**Review by siebeprojects — APPROVED** [review]

**Review by Fbada006 — APPROVED** [review]

**Pull request merged**